docs/customresources/staticroute.md

```diff
@@ -0,0 +1,356 @@
+            # the timeout for the idle upstream of downstream connection, seconds. 0 means forever, unspecified default 1h.
+            idle_timeout: <int>
+            # retres define how many retries to upstream with failed (50x code) requests, number. Default 1.
+            retries: <int>
+          # Optional
+          path:
+            # Rewrites path with the regex and substitution patterns. 
+            rewrite:
+              pattern: <string> # path regex pattern
+              substitution: <string> # path substitution pattern.
+          # Optional
+          cors:
+            # allowed origins returned in Access-Control-Allow-Origin header
+            # the list of domain names
+            # Note - regexs other than the wildcard ("*") are not supported right now.
+            origins:
+            - "*"
+            # allowed methods to call this endpoint returned in Access-Control-Allow-Methods header
+            # the list of methods
+            methods:
+            - POST
+            - GET
+            # allowed headers returned in Access-Control-Allow-Headers header
+            # the list of headers
+            headers:
+            - Content-Type
+            # allow browser to send credentials, returned with Access-Control-Allow-Credentials header
+            credentials: <true|false>
+            # allowed headers that browser can access returned with  Access-Control-Expose-Headers header
+            # the list of headers
+            expose_headers:
+            - X-Custom-Header1
+            - X-Custom-Header2
+            # how long to cache this CORS information for the browser, returned with Access-Control-Max-Age.
+            max_age: <int>
+        # "redirect" creates HTTP redirect to other endpoint. Mutually exclusive with "route".
+        redirect:
+          # redirect to http or https
+          scheme_redirect: <http|https>
+          # redirect to this hostname
+          host_redirect: <string>
+          # redirects to different port
+          port_redirect: <int>
+          # redirect to different URL path. Mutually exlusive with rewrite_regex.
+          path_redirect: "<string>"
+          # redirect using the rewrite rule. Mutually exlusive with path_redirect.
+          rewrite_regex:
+            # regex
+            pattern: <string>
+            # regex parameters substitution pattern
+            substitution: <string>
+          # response code, by default - Permanent Redirect HTTP 308
+          # available HTTP codes: 301, 302, 303, 307, 308
+          response_code: <int>
+      <http_method>:
+        -- skipped --
+```
+
+## Request matching
+
+We match the incoming request by HOST header, path and HTTP method.
+
+The following fields specify matching.
+
+**hosts** that define the list of HOST headers this configuration applies to. This will create Envoy's VirtualHost with the same name and domain matching. Wildcards are possible, e.g. "*" means "any host".
```

Comment:
```suggestion
**hosts** that define the list of HOST headers this configuration applies to. This will create the Envoy's VirtualHost with the same name and domain matching. Wildcards are possible, e.g. "*" means "any host".
```